src/shogun/lib/DynamicArray.h

```diff
@@ -621,21 +774,13 @@ template <class T> class CDynamicArray :public CSGObject
 		virtual void init()
 		{
 			set_generic<T>();
+			T* head = m_array.data();
+			m_parameters->add_vector(
+			    &head, &num_elements, "array", "Memory for dynamic array.");
```

Comment:
mmm, `m_parameters->add_vector(&head, &(m_array.size()), ....` doesn't work, and it broken the clone progress of DynamicArray. Here is how I implement it.

```
		virtual CSGObject* clone()
		{
			CDynamicArray * cloned = (CDynamicArray*) CSGObject::clone();
			// Since the array vector is registered with
			// current_num_elements as size (see parameter
			// registration) the cloned version has less memory
			// allocated than known to dynarray. We fix this here.
			// cloned->num_elements = cloned->m_array.size();
			return cloned;
		}

	private:

		/** register parameters */
		virtual void init()
		{
			set_generic<T>();
			T* head = m_array.data();
                        // & doesn't accept a rvalue and it seems like auto will make 'size' unrecognisable for add_vector()
			int32_t size = m_array.size();
			m_parameters->add_vector(
			    &head, &size, "array", "Memory for dynamic array.");

			SG_ADD(
			    &free_array, "free_array", "whether array must be freed",
			    MS_NOT_AVAILABLE);
			SG_ADD(&dim1_size, "dim1_size", "Dimension 1", MS_NOT_AVAILABLE);
			SG_ADD(&dim2_size, "dim2_size", "Dimension 2", MS_NOT_AVAILABLE);
			SG_ADD(&dim3_size, "dim3_size", "Dimension 3", MS_NOT_AVAILABLE);
		}
```

Progress stop in here

> 
> [ RUN      ] DynamicObjectArray.array_SpectrumMismatchRBFKernel_clone_equals
> shogun-unit-test(91199,0x7fffe91903c0) malloc: *** error for object 0x10637f125: pointer being freed was not allocated
> *** set a breakpoint in malloc_error_break to debug
> Abort trap: 6

And here is some lldb debug message https://pastebin.mozilla.org/9024822